### ISSUE TITLE
Checking units

### ISSUE THREAD

**Issue by mdeshotel** [Dewberry/ripple1d#123]
Units are now extracted from the source models but we still need to implement a series of checks, conversions, and documentation as necessary through out the ripple process. We need additional handling not only for English vs Metric but for ft vs miles and I suppose meter vs kilometer when it comes to stationing and distance related fields. This extends into issues we have experienced with river stationing being limited to 8 characters and the need to potentially convert from ft to miles to handle character limit. -or other solutions to remedy this character limit.

**Comment by mdeshotel** [comment]
We will raise an error if HEC-RAS units are metric. This should already be documented in the geopackage.

**Comment by slawler** [comment]
Per our discussion, let's raise this in the `create_model_run_normal_depth` endpoint.

**Comment by mdeshotel** [comment]
Currently the subsetted geopackage does not have a metadata table like the source geopackage. Do we want to copy the entire metadata table from the source model geopackage into the subsetted geopackage? Or should we only copy specific keys? @ar-siddiqui 

![Image](https://github.com/user-attachments/assets/9fa5b6b7-e715-4e37-bfa9-093820d5714c)

**Comment by ar-siddiqui** [comment]
I would say specific keys that are relevant to the submodel.

For that information from the source model, that is relevant, it should also be copied. For example source_stac_item_id, sourc_model_geom etc. Basically ability for us to track back to the the source model. But aren't we doing this ripple1d.json ?

**Comment by mdeshotel** [comment]
I also thought we were adding this data to the ripple1d.json but it looks like we are not. This information is in the conflation file but does not get propagated to the ripple1d.json. Should I add this source model metadata to the ripple1d.json now? Including the units? 

If we do put this information in the ripple1d.json can we leave it out of the subsetted geopackage?

**Comment by ar-siddiqui** [comment]
I agree with this approach. The subsetted GPKG being representative solely of the content of sunsetted model while the rippp1d json contains metadata and ripple1d data

**Comment by mdeshotel** [comment]
**Here is an overview of the schema changes I made based on the conversation in this issue and #176 .** 

We now pass `units` from the source_model.gpkg to the conflation.json during `conflate_model`. 

![Image](https://github.com/user-attachments/assets/4b77e4fd-8458-44e6-82cb-2256244a0c21)

`source_model_metadata` is now passed from the conflation.json to the ripple1d.json  during `extract_submodel`. 

![Image](https://github.com/user-attachments/assets/346481a3-ede7-4d75-b59c-bf11d39d3309)

`units` and `steady` (which is a boolean indicating if the source model was steady state) is now passed from the conflation.json to the submodel.gpkg  during `extract_submodel`. This required adding a new metadata table to the submodel.gpkg.

![Image](https://github.com/user-attachments/assets/42e82831-6b2f-4ba5-966d-50db9f2428de)

`create_model_run_normal_depth` reads the ripple1d.json and raises an error if `units` is not `English` or if the `forcing file extension` from the source model does not contains `f` indicating that it is not a steady state model.
